ecpy_hqc_legacy/instruments/drivers/visa/windfreaktech_synthHD_signal_generator.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright 2015-2016 by EcpyHqcLegacy Authors, see AUTHORS for more details.
+#
+# Distributed under the terms of the BSD license.
+#
+# The full license is in the file LICENCE, distributed with this software.
+# -----------------------------------------------------------------------------
+"""Driver for Windfreaktech SynthHD SignalGenerator using VISA library.
+
+"""
+from __future__ import (division, unicode_literals, print_function,
+                        absolute_import)
+
+import re
+from textwrap import fill
+from inspect import cleandoc
+
+from visa import VisaTypeError
+
+from ..driver_tools import (InstrIOError, instrument_property,
+                            secure_communication)
+from ..visa_tools import VisaInstrument
+
+CONVERSION_FACTORS = {'GHz': {'Hz': 1e9, 'kHz': 1e6, 'MHz': 1e3, 'GHz': 1},
+                      'MHz': {'Hz': 1e6, 'kHz': 1e3, 'MHz': 1, 'GHz': 1e-3},
+                      'kHz': {'Hz': 1e3, 'kHz': 1, 'MHz': 1e-3, 'GHz': 1e-6},
+                      'Hz': {'Hz': 1, 'kHz': 1e-3, 'MHz': 1e-6, 'GHz': 1e-9}}
+
+class SynthHD(VisaInstrument):
+    """
+    Generic driver for Agilent PSG SignalGenerator, using the VISA library.
+
+    This driver does not give access to all the functionnality of the
+    instrument but you can extend it if needed. See the documentation of
+    the driver_tools module for more details about writing instruments
+    drivers.
+
+    Parameters
+    ----------
+    see the `VisaInstrument` parameters
+
+    Attributes
+    ----------
+    frequency_unit : str
+        Frequency unit used by the driver. The default unit is 'GHz'. Other
+        valid units are : 'MHz', 'KHz', 'Hz'
+    frequency : float, instrument_property
+        Fixed frequency of the output signal.
+    power : float, instrument_property
+        Fixed power of the output signal.
+    output : bool, instrument_property
+        State of the output 'ON'(True)/'OFF'(False).
+
+    Notes
+    -----
+    This driver has been written for the  but might work for other
+    models using the same SCPI commands.
+
+    """
+    def __init__(self, connection_info, caching_allowed=True,
+                 caching_permissions={}, auto_open=True):
+
+        print(connection_info)
+        super(SynthHD, self).__init__(connection_info, caching_allowed,
+                                         caching_permissions, auto_open)
+        self.channel = 0
+        self.frequency_unit = 'GHz'
+        self.write_termination = ''
+        self.read_termination = ''
+        self.detuneRef = 10.0 # in MHz
```

Comment:
Indeed, it shouldn't. I initially put this here because I was using the synthHD's two ports as signal and reference for a readout. At some point I will add this to the view interface.